lifting-state-up/lifting-state-up.md

```diff
@@ -0,0 +1,221 @@
+
+The `prop` named `onUpdate` references a function that updates a student's data. We can look at `App`'s `updateStudentData` function if we need to see the implementation.
+
+<details>
+
+<summary>Review our current implementation of <code>App</code>.</summary>
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from 'react';
+import StudentList from './components/StudentList';
+
+function App() {
+  const [studentData, setStudentData] = useState([
+    {
+      id: 1,
+      nameData: 'Ada',
+      emailData: 'ada@dev.org',
+      isPresentData: false
+    },
+    {
+      id: 2,
+      nameData: 'Soo-ah',
+      emailData: 'sooah@dev.org',
+      isPresentData: false
+    },
+    {
+      id: 3,
+      nameData: 'Chrissy',
+      emailData: 'chrissy@dev.org',
+      isPresentData: true
+    }
+  ]);
+
+  const updateStudentData = updatedStudent => {
+    const students = studentData.map(student => {
+      if (student.id === updatedStudent.id) {
+        return updatedStudent;
+      } else {
+        return student;
+      }
+    });
+
+    setStudentData(students);
+  }
+
+  return (
+    <main>
+      <h1>Attendance</h1>
+      <StudentList
+        students={studentData}
+        onUpdateStudent={updateStudentData}
+      ></StudentList>
+    </main>
+  );
+}
+
+export default App;
+```
+<!-- prettier-ignore-end -->
+
+</details>
+
+<br/>
+
+<!-- prettier-ignore-start -->
+```js
+const Student = (props) => {
+
+    const onAttendanceButtonClick = () => {
+        const updatedStudent = {
+            id: props.id,
+            nameData: props.name,
+            emailData: props.email,
+            isPresentData: !props.isPresent
+        }
+
+        // Invoke the function passed in through the prop named "onUpate"
+        // This function is referenced by the name "updateStudentData" in App
+        props.onUpdate(updatedStudent);
+    }
+
+    // ... other rendering logic
+}
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                                                                                                                                                                          |
+| --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `const onAttendanceButtonClick = () => { ... }`     | We're creating an event-handling function named `onAttendanceButtonClick`.                                                                                                                                                                                     |
```

Comment:
remind students that we _could_ get event details here, but that we don't care for this example
(also ;)
```suggestion
| `const onAttendanceButtonClick = () => { ... };`     | We're creating an event-handling function named `onAttendanceButtonClick`. We don't need any details about the event, so we don't list `event` as a parameter.                                                                                                                                                                                    |
```